### PR TITLE
Refactor to software defined assets and declarative scheduling

### PR DESCRIPTION
**Pull request by petehunt**
* Refactored to software-defined assets, which will make testing easier, code simpler and the UI prettier
* Switched to declarative scheduling

Screen Shot 2023-02-17 at 1 53 42 PM

### CODE REVIEW

File: loader_logic/dag_abstractions.py [hunk 1]
```diff
@@ -1,53 +1,88 @@
...
 
-    def setup_job(self):
-
-        @asset(name=f"{self.name}_load")
-        def load():
-            asset_loader = self.loader
-            return asset_loader
+    def to_assets(self):
+        @asset(
+            group_name=self.name, name=f"{self.name}_raw_documents", compute_kind="http"
```
Comment (zanderchase) [inline loader_logic/dag_abstractions.py:35]: Love this structure. And compute_kind flag.

File: loader_logic/dag_abstractions.py [hunk 2]
```diff
@@ -1,53 +1,88 @@
...
+        return [raw_documents, documents, vectorstore]
 
-        @job(name=f"{self.name}_job")
-        def load_vectorstore():
-            vectorstore(embedding(load()))
 
-        
-        if self.schedule != 'daily':
-            print('Warning: only daily schedule supported right now')
+class VectorstoreIOManager(IOManager):
```
Comment (zanderchase) [inline loader_logic/dag_abstractions.py:68]: Nice. I was looking at implementing an IOManager next. Could you see a similar one being used but for storing to a vector DB?

**Review by zanderchase — COMMENTED** [review]
Thank you Pete! You are showing me things about Dagster I did not know possible. Seems like you know it pretty well...

File: loader_logic/dag_abstractions.py (hunk 2 above)
Comment (petehunt) [reply loader_logic/dag_abstractions.py:68]: Yeah, 100%

**Pull request merged**